kong/db/strategies/off/init.lua

```diff
@@ -131,6 +131,13 @@ local function page_for_prefix(self, prefix, size, offset, options, follow)
     return nil, err_or_more
   end
 
+  -- special handling for page size 1.
+  -- drop other results
+  if size == 1 and size < res then
```

Comment:
```suggestion
  if size == 1 and size < #res then
```

Reply:
and here, size cannot be 1, because this line ` size = math.max(size, LMDB_MIN_PAGE_SIZE)` makes size large than 1

Reply:
I modifed your patch :

```
diff --git a/kong/db/strategies/off/init.lua b/kong/db/strategies/off/init.lua
index 1f0fd2c30..d7550fdc8 100644
--- a/kong/db/strategies/off/init.lua
+++ b/kong/db/strategies/off/init.lua
@@ -116,6 +116,8 @@ local LMDB_MIN_PAGE_SIZE = 2


 local function page_for_prefix(self, prefix, size, offset, options, follow)
+  local size_1 = size == 1
+
   if not size then
     size = self.connector:get_page_size(options)
   end
@@ -131,6 +133,15 @@ local function page_for_prefix(self, prefix, size, offset, options, follow)
     return nil, err_or_more
   end

+  -- special handling for page size 1.
+  -- drop other results
+  if size_1 and 1 < #res then
+    res = { res[1] }
+    err_or_more = true
+  end
+
+  ngx.log(ngx.INFO, "+++ size:", size)
+
   local ret = {}
   local ret_idx = 0
   local schema = self.schema
```

Reply:
solved in lua-resty-lmdb.